### PR TITLE
make: clippy error on warnings

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

Fix our CI to reject code with clippy warnings.

### Testing Strategy

Running locally. This should also fail in CI until the other PR is merged.

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by lschuermann — APPROVED** [review]

**Comment by lschuermann** [comment]
Marking blocked on #4105.

**Comment by lschuermann** [comment]
No need to rebase this once #4105 is in, the merge queue should test the merged result with master. Actually, this will be a great test for exactly this behavior. :smile:

**Review by ppannuto — APPROVED** [review]
Well, let's see if Leon's right :)

**Pull request merged**
